pkg/injector/bootstrap.go

```diff
@@ -95,9 +104,122 @@ func (wh *mutatingWebhook) marshalAndSaveBootstrap(name, namespace string, confi
 			bootstrap.EnvoyXDSCACertFile:                  cert.GetTrustedCAs(),
 			bootstrap.EnvoyXDSCertFile:                    cert.GetCertificateChain(),
 			bootstrap.EnvoyXDSKeyFile:                     cert.GetPrivateKey(),
+			signingIssuerID:                               []byte(cert.GetSigningIssuerID()),
+			validatingIssuerID:                            []byte(cert.GetValidatingIssuerID()),
 		},
 	}
 
 	log.Debug().Msgf("Creating bootstrap config for Envoy: name=%s, namespace=%s", name, namespace)
 	return wh.kubeClient.CoreV1().Secrets(namespace).Create(context.Background(), secret, metav1.CreateOptions{})
 }
+
+// NewBootstrapSecretRotator returns a new bootstrap secret rotator.
+func NewBootstrapSecretRotator(kubeController k8s.Controller, certManager *certificate.Manager, checkInterval time.Duration) *BootstrapSecretRotator {
+	return &BootstrapSecretRotator{
+		kubeController: kubeController,
+		certManager:    certManager,
+		checkInterval:  checkInterval,
+	}
+}
+
+// getBootstrapSecrets returns the bootstrap secrets stored in the informerCollection's store.
+func (b *BootstrapSecretRotator) getBootstrapSecrets() []*corev1.Secret {
+	secrets := b.kubeController.ListSecrets()
+	var bootstrapSecrets []*corev1.Secret
+
+	for _, secret := range secrets {
+		// finds bootstrap secrets
+		if strings.Contains(secret.Name, bootstrapSecretPrefix) {
+			bootstrapSecrets = append(bootstrapSecrets, secret)
+		}
+	}
+	return bootstrapSecrets
+}
+
+// rotateBootstrapSecrets updates the bootstrap secret by getting the current or issuing a new certificate.
+func (b *BootstrapSecretRotator) rotateBootstrapSecrets(ctx context.Context) {
+	bootstrapSecrets := b.getBootstrapSecrets()
+	for _, secret := range bootstrapSecrets {
+		bootstrapCert, err := getCertFromSecret(secret)
+		if err != nil {
+			log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrObtainingCertFromSecret)).
+				Msgf("Error getting cert %s from secret %s/%s", bootstrapCert, secret.Namespace, secret.Name)
+		}
+
+		if !b.certManager.ShouldRotate(bootstrapCert) {
+			continue
+		}
+
+		issuedCert, err := b.certManager.IssueCertificate(bootstrapCert.CommonName.String(), certificate.Internal, certificate.FullCNProvided())
+		if err != nil {
+			log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrRotatingCert)).Msgf("Error rotating cert %s", issuedCert)
+		}
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I think `RetryOnConflict` needs to include the "get" operation. Without a call to get the updated version of the secret, the conflict error will occur on every retry. 

Here is a good example of `RetryOnConflict` from the [docs](https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict):

```go
err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
    // Fetch the resource here; you need to refetch it on every try, since
    // if you got a conflict on the last update attempt then you need to get
    // the current version before making your own changes.
    pod, err := c.Pods("mynamespace").Get(name, metav1.GetOptions{})
    if err != nil {
        return err
    }

    // Make whatever updates to the resource are needed
    pod.Status.Phase = v1.PodFailed

    // Try to update
    _, err = c.Pods("mynamespace").UpdateStatus(pod)
    // You have to return err itself here (not wrapped inside another error)
    // so that RetryOnConflict can identify it correctly.
    return err
})
if err != nil {
    // May be conflict if max retries were hit, or may be something unrelated
    // like permissions or a network error
    return err
}
```

To implement a similar pattern, we will need to add a `GetSecret` method in `k8s/client.go`. After getting the secret and the certificate we will also need to check if the certificate still needs to be updated since it might have already been updated by another injector.